R/s3.R

```diff
@@ -329,3 +329,26 @@ s3_delete <- function(uri) {
     s3_object(uri)$delete()
     log_debug('Deleted %s', uri)
 }
+
+
+#' Sets tags on s3 object overwriting all existing tags. Note: tags and metadata tags are not the same.
+#' Calls this function under the hood: https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3.html#S3.Client.put_object_tagging
+#' @param uri string, URI of an S3 object, should start with \code{s3://}, then bucket name and object key
+#' @param tags string, of the format list('my_first_key' = 'my_first_value', 'my_second_key' = 'my_second_value') where keys are the tag names and values are the tag values. Keys and values must be strings.
```

Comment:
Instead of "list with key/values", I would describe this as a "named character vector".

  c(my_first_key = 'my_first_value', my_second_key = 'my_second_value')

works just as well and is typed a little bit stronger.